[pt] Add content/pt/docs/concepts/instrumentation/code-based.md

janssenlima opened the pull request:
Add localization instrumentation code-based.

Issue: https://github.com/open-telemetry/opentelemetry.io/issues/4922

emdneto commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+deve identificar exatamente o que está sendo instrumentado - se você está
+desenvolvendo uma biblioteca, por exemplo, você deve nomeá-la com o nome da sua
+biblioteca (por exemplo `com.example.myLibrary`), pois esse nome será usado como
+namespace todos os eventos de trecho ou métricas que for produzidos. Além disso,
```
```suggestion
namespace todos os eventos de trecho ou métricas que forem produzidos. Além disso,
```

emdneto replied:
```suggestion
_namespace_ em todos os eventos de trecho ou métricas que for produzidos. Além disso,
```

emdneto commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+
+Para criar rastros ou métricas, primeiro você precisa criar um provider tracer
+e/ou meter. Geralmente, recomendamos que o SDK forneça um único provider padrão
+para esses objetos. Em seguida, você obtem uma instância tracer ou meter desse
```
```suggestion
para esses objetos. Em seguida, você obtém uma instância tracer ou meter desse
```

EzzioMoreira commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+Depois de criar os dados de telemetria, você vai querer enviá-los para algum
+lugar. OpenTelemetry oferece suporte a dois métodos principais de exportação de
+dados do seu processo para um backend de análise, diretamente de um processo ou
+através de um proxy através do [OpenTelemetry Collector](/docs/collector).
```
```suggestion
por meio de um proxy através do [OpenTelemetry Collector](/docs/collector).
```

Só pra não ficar muito repetitivo a palavra através.

EzzioMoreira commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+## Exportar dados
+
+Depois de criar os dados de telemetria, você vai querer enviá-los para algum
+lugar. OpenTelemetry oferece suporte a dois métodos principais de exportação de
```
```suggestion
lugar. O OpenTelemetry oferece suporte a dois métodos principais de exportação de
```
Tivermos um papo e definimos adicionar pronome.

emdneto reviewed (approved):
Thanks

emdneto commented:
/fix:all

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+
+## Configurar o SDK do OpenTelemetry
+
+Se você está desenvolvendo um processo de serviço, também precisará configurar o
```
```suggestion
Se você está desenvolvendo um serviço, também precisará configurar o
```

Em inglês está "service process", então a sua tradução está correta, mas eu acho que ficou um pouco confuso porque "processo" pode significar muitas coisas. Nesse case, eu acho que o texto original quis usar no sentido de processo do SO, mas isso não ficou claro para mim em português.

Acho que dá para tirar o "processo" sem alterar o sentido da frase.

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+
+Se você está desenvolvendo um processo de serviço, também precisará configurar o
+SDK com as opções adequadas para exportar seus dados de telemetria para um
+backend de análise. Recomendamos que esta configuração seja feita
```
```suggestion
_backend_ de análise. Recomendamos que esta configuração seja feita
```

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+Além disso, existem opções de ajuste específicas para cada linguagem de
+programação que você pode utilizar.
+
+## Criar dados de telemetria
```
```suggestion
## Crie dados de telemetria
```

Algumas outras opções poderiam ser

```suggestion
## Gere dados de telemetria
```

```suggestion
## Emita dados de telemetria
```

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+
+## Criar dados de telemetria
+
+Depois de configurar a API e o SDK, você estará livre para criar eventos de
```
```suggestion
Depois de configurar a API e o SDK, você estará pronto para criar eventos de
```

"be free" pode ser traduzido como "estar livre", mas nesse caso está sendo usado mais como uma expressão no sentido de dar permissão, ou deixar alguma coisa acontecer (por exemplo, "feel free to come by at any time")

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+## Criar dados de telemetria
+
+Depois de configurar a API e o SDK, você estará livre para criar eventos de
+rastro e métricas usando os objetos tracer e meter obtidos do provider. Utilize
```
```suggestion
rastros e métricas usando os objetos _tracer_ e _meter_ obtidos do _provider_. Utilize
```

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+Depois de configurar a API e o SDK, você estará livre para criar eventos de
+rastro e métricas usando os objetos tracer e meter obtidos do provider. Utilize
+bibliotecas de instrumentação para suas dependências -- confira o
+[registry](/ecosystem/registry/) ou o repositório da sua linguagem de
```
Hum....."registry" é mais um termo que eu não sei se é comumente utilizado em inglês ou se deveria ser traduzido.

E se for para traduzir eu não sei qual termo utilizar. "Registro" não perece muito certo, "arquivo" (de "archive") é meio ambíguo.

@edsoncelio @EzzioMoreira alguma sugestão?

janssenlima replied:
https://opentelemetry.io/ecosystem/registry/ é um mecanismo de busca.

lgfa29 replied:
Ah sim, minha pergunta foi mais no sentido de se devemos ou não traduzir a palavra "registry" 🙂

emdneto replied:
Eu colocaria `OpenTelemetry Registry`

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+[registry](/ecosystem/registry/) ou o repositório da sua linguagem de
+programação para mais informações sobre essas bibliotecas.
+
+## Exportar dados
```
```suggestion
## Exporte dados
```

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+
+Depois de criar os dados de telemetria, você vai querer enviá-los para algum
+lugar. O OpenTelemetry oferece suporte a dois métodos principais de exportação
+de dados do seu processo para um backend de análise, diretamente de um processo
```
```suggestion
de dados do seu processo para um _backend_ de análise, diretamente de um processo
```

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+de dados do seu processo para um backend de análise, diretamente de um processo
+ou por meio de um proxy através do [OpenTelemetry Collector](/docs/collector).
+
+A exportação em processo exige que você importe e dependa de um ou mais
```
```suggestion
Exportar a partir de um processo exige que você importe e dependa de um ou mais
```

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+de análise de telemetria, como Jaeger ou Prometheus. Além disso, o OpenTelemetry
+suporta um protocolo conhecido como `OTLP`, que é compatível com todos os SDKs
+do OpenTelemetry. Esse protocolo pode ser utilizado para enviar dados ao
+OpenTelemetry Collector, um processo binário que pode ser executado como proxy,
+sidecar ou em um host separado. O Collector pode então ser configurado para
```
```suggestion
OpenTelemetry Collector, um processo binário que pode ser executado como _proxy_,
_sidecar_ ou em um _host_ separado. O Collector pode então ser configurado para
```

lgfa29 commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+sidecar ou em um host separado. O Collector pode então ser configurado para
+encaminhar e exportar esses dados para ferramentas de análise de sua escolha.
+
+Além das ferramentas open source como Jaeger ou Prometheus, uma lista crescente
```
```suggestion
Além das ferramentas _open source_ como Jaeger ou Prometheus, uma lista crescente
```

lgfa29 reviewed (commented):
Ótimo trabalho @janssenlima!

Deixei algumas ideias e sugestões 🙂

chalin commented:
@emdneto et al.: is this ready to be merged (and have any remaining comments addressed in a followup PR)?

janssenlima commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+---
+title: Baseada em código
+description:
+  Informe-se sobre as etapas essenciais na configuração da instrumentação
```
No contexto dessa documentação é para se informar. Não tem nenhum exemplo nessa documentação que justifique um aprendizado somente pela leitura. Mas, vamos aguardar mais opiniões.

janssenlima commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+default_lang_commit: fe0c3f68902aeb6e7584ffc32c6c8964d7a89e6e
+---
+
+## Importar a API e o SDK do OpenTelemetry
```
Não sei se você entendeu essa parte da documentação. Ela é uma sequência de etapas com orientações. A voz ativa é clara, direta, e atribui claramente a responsabilidade pela ação. É útil quando você quer enfatizar quem deve realizar a tarefa. Não é o caso da documentação, onde não sabemos que estará lendo-a. Coloquei no infinitivo porque o foco da documentação é na ação a ser realizada, não em quem irá realizar.

lgfa29 replied:
Ah sim, mas nesse caso o foco é realmente quem está lendo, por isso que diversas frases começam com "você precisará..." ou "você deve....". , então a voz ativa deixa claro que você (como pessoa lendo a página) precisa executar essas ações. 

É por isso também que documentação técnica em geral é escrita utilizando a voz ativa (https://developers.google.com/tech-writing/one/active-voice).

emdneto commented:
@chalin I'm moving this to draft while @janssenlima address the suggestions

janssenlima commented on content/pt/docs/concepts/instrumentation/code-based.md:
```diff
@@ -0,0 +1,70 @@
+title: Baseada em código
+description:
+  Informe-se sobre as etapas essenciais na configuração da instrumentação
+  baseada em código
```
Aqui eu iria mais além de não colocar apenas uma tradução, mas mudar o termo.
Ouvimos muito instrumentação manual e instrumentação automática.
Eu sugiro trocarmos `zero-code `para `Instrumentação automática` e `code-based` para `Instrumentação manual`. Faz sentido para vocês?

marcelobaptista replied:
Que tal autoinstrumentação?

emdneto replied:
Acho que nesse caso baseado em código pode ser substituido por: 
instrumentação manual ou programática.

emdneto replied:
>Eu sugiro trocarmos zero-code para Instrumentação automática e code-based para Instrumentação manual. 

Estou de acordo com isso

edsoncelio replied:
> Aqui eu iria mais além de não colocar apenas uma tradução, mas mudar o termo. Ouvimos muito instrumentação manual e instrumentação automática. Eu sugiro trocarmos `zero-code `para `Instrumentação automática` e `code-based` para `Instrumentação manual`. Faz sentido para vocês?

+ 1 pra essa sugestão.
Podemos padronizar o uso (e adicionar no glossário).

janssenlima commented:
> @chalin I will move this PR to draft while @janssenlima addresses the suggestions. There are important fixes that we missed at the first look

@emdneto, fiz os ajustes. Só tem uma thread ali em discussão para fechar a documentação.

janssenlima commented:
@emdneto, fiz todas as correções.

emdneto commented:
/fix:all
